plugins/base/src/test/kotlin/signatures/SignatureTest.kt

```diff
@@ -185,6 +185,31 @@ class SignatureTest : BaseAbstractTest() {
         }
     }
 
+    @Test
+    fun `fun with definitely non-nullable types`() {
+        val source = source("fun <T> elvisLike(x: T, y: T & Any): T & Any = x ?: y")
+        val writerPlugin = TestOutputWriterPlugin()
+
+        testInline(
+            source,
+            configuration,
+            pluginOverrides = listOf(writerPlugin)
+        ) {
+            renderingStage = { _, _ ->
```

Comment:
can you also add asserts for documentable model level? i.e make sure it resolves to `DefinitelyNonNullable` and assert DRI for `Any`. Here or maybe `DRITest` would be a better place